src/scalar/text.ts

```diff
@@ -0,0 +1,47 @@
+import { Utf8 as ArrowString } from '@apache-arrow/esnext-esm';
+
+import { isInvalid, NULL_VALUE } from './util.js';
+
+export class Text {
+  private _valid = false;
+  private _value = '';
+
+  public constructor(v: unknown) {
+    this.value = v;
+    return this;
+  }
+
+  public get dataType() {
+    return new ArrowString();
+  }
+
+  public get valid(): boolean {
+    return this._valid;
+  }
+
+  public get value(): string {
+    return this._value;
+  }
+
+  public set value(value: unknown) {
+    if (isInvalid(value)) {
+      this._valid = false;
+      return;
+    }
+    if (typeof value === 'string' || value instanceof String) {
```

Comment:
This means we only support settings strings to the string scalar.
We should support more types as in:

Reply:
I think the equivalent to `Stringer` in Go is to check `Object.hasOwn(object1, 'toString')`, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn